### ISSUE TITLE
Saving content script in dev mode for Manifest V2 throws a TypeError

### ISSUE THREAD

**Issue by hikiko4ern** [wxt-dev/wxt#420]
### Describe the bug

Saving the content script in dev mode for Manifest V2 (e.g. Firefox) throws

```
file://<path>/node_modules/wxt/dist/cli.js:2177
        return [key, value.sort()];
                           ^

TypeError: Cannot assign to read only property '0' of object '[object Array]'
    at Array.sort (<anonymous>)
    at file://<path>/node_modules/wxt/dist/cli.js:2177:28
    at Array.map (<anonymous>)
    at hashContentScriptOptions (file://<path>/node_modules/wxt/dist/cli.js:2175:34)
    at file://<path>/node_modules/wxt/dist/cli.js:2486:22
    at Array.reduce (<anonymous>)
    at addEntrypoints (file://<path>/node_modules/wxt/dist/cli.js:2485:51)
    at generateManifest (file://<path>/node_modules/wxt/dist/cli.js:2282:3)
    at async rebuild (file://<path>/node_modules/wxt/dist/cli.js:2665:65)
    at async file://<path>/node_modules/wxt/dist/cli.js:3144:37
```

at this line

https://github.com/wxt-dev/wxt/blob/bb022646e320114c685acea0eb2d8e74d0f84f6c/src/core/utils/content-scripts.ts#L38

This error is not thrown when using MV3.

### To Reproduce

[wxt-sort-readonly.zip](https://github.com/wxt-dev/wxt/files/14166489/wxt-sort-readonly.zip)

1. Install dependencies: `pnpm i`
2. Start dev mode for MV2: `pnpm dev:firefox` or `pnpm dev --mv2`
3. Save file `entrypoints/content.ts`

### Environment

```
  System:
    OS: Linux 5.15 Debian GNU/Linux trixie/sid
    CPU: (32) x64 AMD Ryzen 9 5950X 16-Core Processor
    Memory: 59.91 GB / 70.67 GB
    Container: Yes
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.11.0 - /mnt/wslg/runtime-dir/fnm_multishells/29666_1707135069144/bin/node
    npm: 10.2.4 - /mnt/wslg/runtime-dir/fnm_multishells/29666_1707135069144/bin/npm
    pnpm: 8.15.1 - /mnt/wslg/runtime-dir/fnm_multishells/29666_1707135069144/bin/pnpm
    bun: 1.0.26 - ~/.bun/bin/bun
  Browsers:
    Chrome: 120.0.6099.216
    Chromium: 120.0.6099.199
  npmPackages:
    wxt: ^0.16.0 => 0.16.3
```

### Additional context

This error seems to occur because the `immer.produce` used here
https://github.com/wxt-dev/wxt/blob/21636defdaeb41afc3a34ad77d847f60962212b4/src/core/utils/manifest.ts#L116
makes the manifest immutable.

Using `.toSorted()` instead of `.sort()` solves the problem, but I'm not sure what consequences this might have.

**Comment by aklinker1** [comment]
Interesting, good catch. The same code runs in MV3, so it's surprising that it doesn't cache there as well. I'll look into that too.

**Comment by aklinker1** [comment]
Fix released in [`v0.16.4`](https://github.com/wxt-dev/wxt/releases/tag/v0.16.4)
